.github/workflows/update-npm-dependencies.yml

```diff
@@ -0,0 +1,73 @@
+name: update-npm-dependencies
+
+on:
+  # Manual run
+  workflow_dispatch:
+  pull_request:
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+
+jobs:
+  update-npm-dependencies:
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout aspnetcore repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'dotnet/aspnetcore'
+          path: 'main'
+
+      - name: Checkout Node-Externals repository
+        uses: actions/checkout@v4
+        with:
+          repository: 'dotnet/Node-Externals'
+          path: 'node-externals'
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+
+      - name: Run npm install to update package-lock.json
+        shell: pwsh
+        id: npm-install-script
+        run: ./main/.github/workflows/update-npm-dependencies.ps1
+
+      - name: Create PR to update the offline package cache
+        # if: steps.update-npm-dependencies.outputs.changes == 'true'
+        id: update-offline-package-cache
+        uses: dotnet/actions-create-pull-request@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: ./node-externals/
+          commit-message: 'Update the NPM offline cache'
+          title: '[Infrastructure] Update the NPM offline cache'
+          body: 'Updates the npm offline cache to match the associated package-lock.json file in dotnet/aspnetcore.'
+          labels: 'Type: Dependency Update :arrow_up_small:'
```

Comment:
I wonder if this can be done by label ID so it doesn't get out of sync if someone tweaks the emoji or something?